File: JobApplicationMicroservice/src/main/java/com/licenta/jobapplicationmicroservice/business/model/Job.java
package com.licenta.jobapplicationmicroservice.business.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class Job {

    private Long id;
    private String title;
    private Long recruiterId;
    private Long companyId;
    private Integer cityId;
    private Integer countryId;
    private Integer workTypeId; // on-site, remote, hybrid
    private Integer jobTypeId; // full-time, part-time, seasonal, temporary
    private Integer experienceLevelId; // internship, entry-level, associate, mid-senior level, director
    private Integer companyIndustryId; // IT, etc
    private Integer industryId; // IT, etc
    private String description;
    private String postedAt;
}

File: NewsfeedMicroservice/src/main/kotlin/com/licenta/newsfeedmicroservice/business/model/Job.kt
package com.licenta.newsfeedmicroservice.business.model

import java.io.Serializable

data class Job(
    val id: Long,
    val title: String,
    val companyId: Long,
    val cityId: Int,
    val countryId: Int,
    val workTypeId: Int, // on-site, remote, hybrid
    val jobTypeId: Int, // full-time, part-time, seasonal, temporary
    val experienceLevelId: Int, // internship, entry-level, associate, mid-senior level, director
    val companyIndustryId: Long, // IT, etc
    val industryId: Int, // IT, etc
    val description: String,
    val postedAt: String
): Serializable